### ISSUE TITLE
Use jsonvalidate

### ISSUE THREAD

**Issue by ateucher** [ropenscilabs/geojsonlint#1]
Hey @sckott I've implemented using the jsonvalidate package to use the geojson schema [here](https://github.com/ateucher/geojsonlint/tree/add-jsonvalidate). I need to flesh out the tests, documentation, and add some examples, but it seems to work.

One odd thing: The core function (`validate_geojson`) for validating against a schema is created using a closure (`jsonvalidate::jsonvalidator`) that assigns the schema into a V8 context inside that function. I could only get it to work by generating `validate_geojson` in `.onLoad` (see commit [here](https://github.com/ateucher/geojsonlint/commit/9cf5c6a458ba901b69b5b503d3a01e6172fc6fd9)). Otherwise, if it is just created in a .R source file, when you try to call any functions that use it, it can't find the schema - I don't fully understand it, but I assume it has something to do with it not being loaded into the V8 context properly unless it's done during package load...

**Comment by sckott** [comment]
hmm, i'll have a look and see - if we do it in onload - is it fast at least?

**Comment by ateucher** [comment]
yep

``` r
library(geojsonlint)
library(microbenchmark)

california <- readLines("https://raw.githubusercontent.com/glynnbird/usstatesgeojson/master/california.geojson")
```

``` r
microbenchmark(geojson_hint(california),
               geojson_validate(california))
```

    ## Unit: milliseconds
    ##                          expr        min         lq       mean     median
    ##      geojson_hint(california) 127.446991 131.841991 137.751790 133.542034
    ##  geojson_validate(california)   5.682101   6.105148   6.853558   6.390921
    ##          uq       max neval cld
    ##  138.169245 226.96137   100   b
    ##    6.781733  21.23411   100  a

**Comment by sckott** [comment]
nice, I just meant the actual onload step vs. without it happening at all - it's probably not noticeable

**Comment by sckott** [comment]
tested out your branch, works well, i think that makes sense to use onload, LGTM

**Comment by ateucher** [comment]
Oh duh, right :)

Great! Do you want a PR now, or wait for more tests and documentation (which I may not be able to get to for a while)?

**Comment by sckott** [comment]
now is good, no rush to get to cran

**Comment by sckott** [comment]
closed via #3
